octodns/record/__init__.py

```diff
@@ -1256,13 +1256,40 @@ class NsRecord(_ValuesMixin, Record):
 
 
 class PtrValue(_TargetValue):
-    pass
 
+    @classmethod
+    def validate(cls, values, _type):
+        if not isinstance(values, list):
+            values = [values]
+
+        reasons = []
+
+        if not values:
+            reasons.append('missing values')
+
+        for value in values:
+            reasons.extend(super(PtrValue, cls).validate(value, _type))
+
+        return reasons
 
-class PtrRecord(_ValueMixin, Record):
+    @classmethod
+    def process(cls, values):
+        return [super(PtrValue, cls).process(v) for v in values]
+
+
+class PtrRecord(_ValuesMixin, Record):
     _type = 'PTR'
     _value_type = PtrValue
 
+    # This is for backward compatibility with providers that don't support
+    # multi-value PTR records.
+    @property
+    def value(self):
+        if len(self.values) == 1:
+            return self.data['value']
+
+        raise AttributeError("Multi-value PTR record has no attribute 'value'")
```

Comment:
Will have to think about this a bit more. Not quite feeling right as it seems like the result is that providers that don't support it just blow up on apply if you put multiple values in a `PTR`, but don't indicate that anything is wrong during plan and would just show that it's going to change...

My second thought (first didn't pan out when thinking it through) is that we might have to implement something like `SUPPORTS_MULTIVALUE_PTR` and have the manager check that anytime is sees multiple values in a `PTR` during the `populate` targets phase during planning. It that works then things would just proceed normally if everything is single value or if there's multi-value PTRs and all the target support them. If there's multi-value and one or more of the targets doesn't then it'd blow up immediately with an error saying as much...

Reply:
As mentioned over in https://github.com/octodns/octodns/pull/755#pullrequestreview-730311028 the blow up when things aren't supported isn't really how octoDNS currently works so it probably doesn't make sense to make that sort of change now. The current behavior is more that it WARNS and continues as was done in https://github.com/octodns/octodns/pull/756.

I think that's the right tact here. I don't think we need to blow up when there's multiple values, instead we should pick the first one and use that after warning that's what is going to happen.

Reply:
Ok makes sense. Let me work on this.

Reply:
See if you like it now.

Reply:
Hold on, this is not going to work. Sync will keep wanting to make an update since config's multi-value differs from target's single-value. Let me re-think this.

Reply:
May be a case where `_include_change` is useful. You can grep around for examples of what it's currently doing.

Reply:
`_include_change` in base provider is my primary candidate right now. But it relies on providers calling `super(..)._include_change`. A new provider in future can easily omit the super call and sync will keep showing plans with multiple values.

Reply:
I've added something around _include_change. Tested with mutli value support turned on and off, seems to work as expected.